### ETL: автоматизация подготовки данных (семинары).  
#### Урок 2. Введение в подготовку данных для аналитиков. Таблицы фактов и таблицы измерений.

#### 1. Скачайте датасет fifаs2.сsv. Проанализируйте его и определите, какие данные являются неполными. Удалите ненужные колонки и недостающие значения.

#### 2. Найдите в датафрейме полные дубликаты и удалите их. Значения могут быть одинаковыми, но написаны по-разному. Например, может отличаться размер регистра (заглавные и строчные буквы). Особое внимание уделить колонке с названиями команд.
#### 3. Напишите функцию, которая добавит колонку с разбиением возраста по группам: до 20, от 20 до 30, от 30 до 36 и старше 36. Посчитайте количество футболистов в каждой категории.

#### Описание dataset fifa_s2:

#### Включает в себя файл fifa_s2.csv, набор данных о большинстве футболистов по всему миру.
#### Каждая строчка наборов данных содержит следующие поля:

   - ID — идентификационный номер футболиста;

   - Name — имя;

   - Age — возраст;

   - Nationality — национальность;

   - Overall — в целом;

   - Potential — потенциал игрока (насколько высокого уровня он может достичь);

   - Club — клуб футболиста;

   - Value — значение в евро;

   - Wage — заработная плата;

   - Preferred Foot — предпочтительная нога;

   - International Reputation — международный рейтинг.
   
   - Skill Moves — владение навыками.
   
   - Position — позиция игрока на поле.
   
   - Joined — год вступления в клуб.
   
   - Contract Valid Until — дата окончания контракта.
   
   - Height — рост игрока.
   
   - Weight — вес игрока.
   
   - Release Clause — плата за освобождение.


### Задача №1. Скачайте датасет fifаs2.сsv.  
### Проанализируйте его и определите, какие данные являются неполными. Удалите ненужные колонки и недостающие значения.

In [206]:
import pandas as pd                                # Импорт модулей для анализа данных
import numpy as np
import seaborn as sns

import matplotlib.pyplot as plt
import matplotlib.mlab as mlab
import matplotlib
plt.style.use('ggplot')
from matplotlib.pyplot import figure

%matplotlib inline
matplotlib.rcParams['figure.figsize'] = (12,8)

pd.options.mode.chained_assignment = None

In [207]:
df = pd.read_csv('fifa_s2.csv')
df.head()

ID       Name  Age  Nationality  Overall  Potential  \
0   1179  G. Buffon   40        Italy       88         88   
1   5479   Casillas   37        Spain       82         82   
2   8205      Braga   35     Portugal       70         70   
3   9014  A. Robben   34  Netherlands       84         84   
4  19521  J. Ibehre   35      England       61         61   

                  Club    Value  Wage Preferred Foot  \
0  Paris Saint-Germain   4000.0    77          Right   
1             FC Porto   1500.0    10           Left   
2              CD Aves    750.0     5          Right   
3    FC Bayern München  15500.0   110           Left   
4     Cambridge United    120.0     2          Right   

   International Reputation  Skill Moves Position  Joined  \
0                       4.0          1.0       GK    2018   
1                       4.0          1.0      NaN    2015   
2                       1.0          3.0      LCM    2017   
3                       4.0          4.0       RM    2009   
4                       1.0          2.0       ST    2017   

  Contract Valid Until    Height  Weight  Release Clause  
0           01/01/2019  6.333333   203.0          7400.0  
1           01/01/2019  6.083333   185.0          3000.0  
2           01/01/2019  5.916667   165.0          1600.0  
3           01/01/2019  5.916667   176.0         25600.0  
4           01/01/2019  6.333333   203.0           210.0

In [208]:
df.info()                                 # Информация о наборе данных, включая размер набора данных, имя столбца, тип
df.columns.values

df_numeric = df.select_dtypes(include=[np.number])  # Столбцы имеющие числовые значения.
numeric_cols = df_numeric.columns.values
print('Столбцы имеют числовые значения:','\n',numeric_cols)

df_non_numeric = df.select_dtypes(exclude=[np.number])    # Столбцы не имеющие числовые значения.
non_numeric_cols = df_non_numeric.columns.values
print('Столбцы имеют не числовые значения:','\n',non_numeric_cols)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2399 entries, 0 to 2398
Data columns (total 18 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   ID                        2399 non-null   int64  
 1   Name                      2399 non-null   object 
 2   Age                       2399 non-null   int64  
 3   Nationality               2392 non-null   object 
 4   Overall                   2399 non-null   int64  
 5   Potential                 2399 non-null   int64  
 6   Club                      2158 non-null   object 
 7   Value                     2142 non-null   float64
 8   Wage                      2399 non-null   int64  
 9   Preferred Foot            2399 non-null   object 
 10  International Reputation  2389 non-null   float64
 11  Skill Moves               2393 non-null   float64
 12  Position                  2378 non-null   object 
 13  Joined                    2399 non-null   int64  
 14  Contract

Благодаря предварительному наблюдению за данными, эта выборка данных содержит
в общей сложности 2399 строку * 18 столбцов данных.  
Имеются столбцы с числовымы и не числовыми данными.  
Наблюдения показали, что имеются 8 столбцов, которые содержат нулевые значения:

#### Столбцы с нулевыми значениями

In [209]:
print(df['Nationality'].isnull().value_counts())
print('-'*50)
print(df['Club'].isnull().value_counts())
print('-'*50)
print(df['Value'].isnull().value_counts())
print('-'*50)
print(df['International Reputation'].isnull().value_counts())        # Показать столбцы с нулевыми значениями
print('-'*50)
print(df['Skill Moves'].isnull().value_counts())
print('-'*50)
print(df['Position'].isnull().value_counts())
print('-'*50)
print(df['Contract Valid Until'].isnull().value_counts())
print('-'*50)
print(df['Release Clause'].isnull().value_counts())
print('-'*50)

False    2392
True        7
Name: Nationality, dtype: int64
--------------------------------------------------
False    2158
True      241
Name: Club, dtype: int64
--------------------------------------------------
False    2142
True      257
Name: Value, dtype: int64
--------------------------------------------------
False    2389
True       10
Name: International Reputation, dtype: int64
--------------------------------------------------
False    2393
True        6
Name: Skill Moves, dtype: int64
--------------------------------------------------
False    2378
True       21
Name: Position, dtype: int64
--------------------------------------------------
False    2126
True      273
Name: Contract Valid Until, dtype: int64
--------------------------------------------------
False    2396
True        3
Name: Release Clause, dtype: int64
--------------------------------------------------


#### Описательная статистика.

In [210]:
df.describe()

ID          Age      Overall    Potential          Value  \
count    2399.000000  2399.000000  2399.000000  2399.000000    2142.000000   
mean   212033.680700    25.213005    69.115048    74.115048    5980.873016   
std     30549.399068     4.633622     8.060195     7.190957   11963.420681   
min      1179.000000    16.000000    47.000000    48.000000      20.000000   
25%    195105.000000    22.000000    64.000000    69.000000     500.000000   
50%    218745.000000    25.000000    69.000000    74.000000    1100.000000   
75%    234576.000000    29.000000    74.000000    79.000000    6000.000000   
max    246608.000000    44.000000    94.000000    95.000000  118500.000000   

              Wage  International Reputation  Skill Moves       Joined  \
count  2399.000000               2389.000000  2393.000000  2399.000000   
mean     24.178825                  1.308497     2.469703  2016.145894   
std      49.552098                  0.687175     0.844531     2.085829   
min       0.000000                  1.000000     1.000000  2000.000000   
25%       1.000000                  1.000000     2.000000  2016.000000   
50%       5.000000                  1.000000     2.000000  2017.000000   
75%      22.000000                  1.000000     3.000000  2018.000000   
max     565.000000                  5.000000     5.000000  2018.000000   

            Height       Weight  Release Clause  
count  2399.000000  2399.000000     2396.000000  
mean      5.955300   166.802782    10690.033498  
std       0.219249    15.241015    21780.029323  
min       5.250000   121.000000       35.000000  
25%       5.833333   154.000000     1100.000000  
50%       6.000000   165.000000     3800.000000  
75%       6.083333   176.000000     7500.000000  
max       6.583333   214.000000   228100.000000

#### Анализ данных.

Средний игрок:  
Возраст ~ 25 лет, Потенциал ~ 74, Зарплата ~ 5980 евро, Навыки владение ~ 2,  
Рост ~ 5,9 футов(176 см), Вес ~ 166 фунтов(75,3 кг), Выходное пособие ~ 10690 евро.

В общей информации о данных видим, значения в столбцах, Value, Wage, Release Clause (50% сильно отличаются от mean)  являются ненормальными.

In [211]:
df[df.Value.isnull()]

ID           Name  Age Nationality  Overall  Potential  \
19     53748  K. Pilkington   44     England       48         48   
48    140222    P. Da Silva   38    Paraguay       72         72   
72    147899        C. Moţi   33     Romania       69         69   
81    153160      R. Raldes   37     Bolivia       70         70   
84    156092      J. Villar   41    Paraguay       77         77   
...      ...            ...  ...         ...      ...        ...   
2388  246403     B. Caicedo   26     Ecuador       72         75   
2389  246404    J. Cifuente   25     Ecuador       68         70   
2390  246405      B. Angulo   22     Ecuador       71         77   
2391  246406     X. Arreaga   23     Ecuador       66         71   
2392  246407        C. Añez   22     Bolivia       63         68   

                  Club  Value  Wage Preferred Foot  International Reputation  \
19    Cambridge United    NaN     1          Right                       1.0   
48                 NaN    NaN     0          Right                       2.0   
72                 NaN    NaN     0          Right                       1.0   
81                 NaN    NaN     0          Right                       1.0   
84                 NaN    NaN     0          Right                       2.0   
...                ...    ...   ...            ...                       ...   
2388               NaN    NaN     0           Left                       1.0   
2389               NaN    NaN     0          Right                       1.0   
2390               NaN    NaN     0          Right                       1.0   
2391               NaN    NaN     0          Right                       1.0   
2392               NaN    NaN     0          Right                       1.0   

      Skill Moves Position  Joined Contract Valid Until    Height  Weight  \
19            1.0       GK    2018           01/01/2023  6.166667   190.0   
48            2.0      RCB    2016                  NaN  5.916667   168.0   
72            2.0      LCB    2016                  NaN  6.000000   172.0   
81            2.0       LB    2016                  NaN  5.916667   172.0   
84            1.0       GK    2016                  NaN  5.916667   187.0   
...           ...      ...     ...                  ...       ...     ...   
2388          3.0       LB    2016                  NaN  5.916667   159.0   
2389          3.0       ST    2016                  NaN  5.833333   170.0   
2390          3.0       ST    2016                  NaN  6.000000   154.0   
2391          2.0       CB    2016                  NaN  6.000000   165.0   
2392          2.0       RB    2016                  NaN  5.750000   143.0   

      Release Clause  
19       4585.060806  
48       4585.060806  
72       4585.060806  
81       4585.060806  
84       4585.060806  
...              ...  
2388     4585.060806  
2389     4585.060806  
2390     4585.060806  
2391     4585.060806  
2392     4585.060806  

[257 rows x 18 columns]

Столбец Value содержит нулевые количественные значения NaN(257).

In [212]:
df[df['Wage'] == 0]

ID          Name  Age Nationality  Overall  Potential Club  Value  \
48    140222   P. Da Silva   38    Paraguay       72         72  NaN    NaN   
72    147899       C. Moţi   33     Romania       69         69  NaN    NaN   
81    153160     R. Raldes   37     Bolivia       70         70  NaN    NaN   
84    156092     J. Villar   41    Paraguay       77         77  NaN    NaN   
89    156713  A. Granqvist   33      Sweden       80         80  NaN    NaN   
...      ...           ...  ...         ...      ...        ...  ...    ...   
2388  246403    B. Caicedo   26     Ecuador       72         75  NaN    NaN   
2389  246404   J. Cifuente   25     Ecuador       68         70  NaN    NaN   
2390  246405     B. Angulo   22     Ecuador       71         77  NaN    NaN   
2391  246406    X. Arreaga   23     Ecuador       66         71  NaN    NaN   
2392  246407       C. Añez   22     Bolivia       63         68  NaN    NaN   

      Wage Preferred Foot  International Reputation  Skill Moves Position  \
48       0          Right                       2.0          2.0      RCB   
72       0          Right                       1.0          2.0      LCB   
81       0          Right                       1.0          2.0       LB   
84       0          Right                       2.0          1.0       GK   
89       0          Right                       2.0          2.0      LCB   
...    ...            ...                       ...          ...      ...   
2388     0           Left                       1.0          3.0       LB   
2389     0          Right                       1.0          3.0       ST   
2390     0          Right                       1.0          3.0       ST   
2391     0          Right                       1.0          2.0       CB   
2392     0          Right                       1.0          2.0       RB   

      Joined Contract Valid Until    Height  Weight  Release Clause  
48      2016                  NaN  5.916667   168.0     4585.060806  
72      2016                  NaN  6.000000   172.0     4585.060806  
81      2016                  NaN  5.916667   172.0     4585.060806  
84      2016                  NaN  5.916667   187.0     4585.060806  
89      2016                  NaN  6.333333   185.0     4585.060806  
...      ...                  ...       ...     ...             ...  
2388    2016                  NaN  5.916667   159.0     4585.060806  
2389    2016                  NaN  5.833333   170.0     4585.060806  
2390    2016                  NaN  6.000000   154.0     4585.060806  
2391    2016                  NaN  6.000000   165.0     4585.060806  
2392    2016                  NaN  5.750000   143.0     4585.060806  

[241 rows x 18 columns]

Столбец Wage содержит количественные значения 0 (241).

Нулевые качественные значения NaN(7) в столбце Nationality заменить на строку 'none'.  
Нулевые качественные значения NaN(241) в столбце Club заменить на строку 'none'.  
Нулевые количественные значения NaN(257) в столбце Value заменить на значение mean.  
Количественные значения 0 (241) в столбце Wage заменить на значение mean.    
Нулевые качественные значения NaN(21) в столбце Position заменить на строку 'none'.      
Нулевые количественные значения NaN(10) в столбце International Reputation заменить на значение min 1.  
Нулевые количественные значения NaN(6) в столбце Skill Moves заменить на значение min 1.  
Нулевые качественные значения NaN(273) в столбце Contract Valid Until заменить на строку 'none'.  
Нулевые количественные значения NaN(3) в столбце Release Clause заменить на значение min 35.

#### Очистка данных.

In [213]:
# Нулевые качественные значения NaN в столбцах заменим на строку 'none'.
df[['Position', 'Nationality', 'Club', 'Contract Valid Until']] = df[['Position', 'Nationality', 'Club', 'Contract Valid Until']]. fillna('none')

# Количественные значения равные 0 (241), заполним mean значением.
df['Wage'].replace(0,df['Wage'].mean(),inplace=True)

# Нулевые количественные значения NaN(257) в столбце Value заменить на значение mean.
df['Value'] = df['Value']. fillna(df['Value'].mean())
df[['Release Clause']] = df[['Release Clause']]. fillna (35)

# Нулевые количественные значения NaN в столбцах заменим на значение min 1. 
df[['Skill Moves', 'International Reputation',]] = df[['Skill Moves', 'International Reputation']]. fillna (1)


In [214]:
df.describe()

ID          Age      Overall    Potential          Value  \
count    2399.000000  2399.000000  2399.000000  2399.000000    2399.000000   
mean   212033.680700    25.213005    69.115048    74.115048    5980.873016   
std     30549.399068     4.633622     8.060195     7.190957   11304.181418   
min      1179.000000    16.000000    47.000000    48.000000      20.000000   
25%    195105.000000    22.000000    64.000000    69.000000     550.000000   
50%    218745.000000    25.000000    69.000000    74.000000    1500.000000   
75%    234576.000000    29.000000    74.000000    79.000000    5980.873016   
max    246608.000000    44.000000    94.000000    95.000000  118500.000000   

              Wage  International Reputation  Skill Moves       Joined  \
count  2399.000000               2399.000000  2399.000000  2399.000000   
mean     26.607794                  1.307211     2.466028  2016.145894   
std      48.895337                  0.686029     0.846664     2.085829   
min       1.000000                  1.000000     1.000000  2000.000000   
25%       2.000000                  1.000000     2.000000  2016.000000   
50%       8.000000                  1.000000     2.000000  2017.000000   
75%      24.178825                  1.000000     3.000000  2018.000000   
max     565.000000                  5.000000     5.000000  2018.000000   

            Height       Weight  Release Clause  
count  2399.000000  2399.000000     2399.000000  
mean      5.955300   166.802782    10676.709154  
std       0.219249    15.241015    21769.659473  
min       5.250000   121.000000       35.000000  
25%       5.833333   154.000000     1100.000000  
50%       6.000000   165.000000     3800.000000  
75%       6.083333   176.000000     7500.000000  
max       6.583333   214.000000   228100.000000

#### Удаление нерелевантных столбцов.

In [215]:
df = df.drop(['Joined','Contract Valid Until'],axis='columns')               # Удалить нерелевантные столбцы
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2399 entries, 0 to 2398
Data columns (total 16 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   ID                        2399 non-null   int64  
 1   Name                      2399 non-null   object 
 2   Age                       2399 non-null   int64  
 3   Nationality               2399 non-null   object 
 4   Overall                   2399 non-null   int64  
 5   Potential                 2399 non-null   int64  
 6   Club                      2399 non-null   object 
 7   Value                     2399 non-null   float64
 8   Wage                      2399 non-null   float64
 9   Preferred Foot            2399 non-null   object 
 10  International Reputation  2399 non-null   float64
 11  Skill Moves               2399 non-null   float64
 12  Position                  2399 non-null   object 
 13  Height                    2399 non-null   float64
 14  Weight  

### Задача №2 Найдите в датафрейме полные дубликаты и удалите их.  
Значения могут быть одинаковыми, но написаны по-разному.  
Например, может отличаться размер регистра (заглавные и строчные буквы). Особое внимание уделить колонке с названиями команд.

In [216]:
# Переводим строковые значения столбцов в нижний регистр
df['Name'] = df['Name'].str.lower()
df['Nationality'] = df['Nationality'].str.lower()
df['Club'] = df['Club'].str.lower()
df['Preferred Foot'] = df['Preferred Foot'].str.lower()
df['Position'] = df['Position'].str.lower()

В названиях клубов имеются значения (1.), пробелы, пропуски, преходы.

In [217]:
# Удаляем пробелы и переходы на новую строку.
df['Name'] = df['Name'].str.replace('\n', '')
df['Name'] = df['Name'].str.replace('\t', ' ')
df['Name'] = df['Name'].str.replace(' {2,}', ' ', regex=True)
df['Name'] = df['Name'].str.strip()

df['Club'] = df['Club'].str.replace('1.', '', regex=True)           # удаляем (1.) в названиях клубов
df['Club'] = df['Club'].str.replace('\n', '')
df['Club'] = df['Club'].str.replace('\t', ' ')
df['Club'] = df['Club'].str.replace(' {2,}', ' ', regex=True)
df['Club'] = df['Club'].str.strip()

#### Находим дубликаты строк.

In [218]:
df["ID"].nunique()                   # Количество уникальных строк

2395

В столбце ID из 2399 значений, 4 неуникальны.

In [219]:
# находим дубликаты по столбцу ID 
df[df.ID.duplicated()]

ID        Name  Age Nationality  Overall  Potential  \
154   168651  i. rakitić   30     croatia       87         87   
304   181872    a. vidal   31       chile       85         85   
687   200466   v. eicher   27     germany       67         69   
2123  241736  y. bisseck   17     germany       61         82   

                  Club    Value   Wage Preferred Foot  \
154       fc barcelona  46500.0  260.0          right   
304       fc barcelona  26000.0  205.0          right   
687   fc heidenheim 46    625.0    4.0           left   
2123           fc köln    500.0    1.0          right   

      International Reputation  Skill Moves Position    Height  Weight  \
154                        4.0          3.0      rcm  6.000000   172.0   
304                        4.0          3.0      cam  5.916667   165.0   
687                        1.0          1.0       gk  6.250000   187.0   
2123                       1.0          2.0       cb  6.333333   183.0   

      Release Clause  
154          95300.0  
304          53300.0  
687           1000.0  
2123          1300.0

In [220]:
df[df['ID'] == 241736]

ID        Name  Age Nationality  Overall  Potential     Club  Value  \
2122  241736  y. bisseck   17     germany       61         82  fc köln  500.0   
2123  241736  y. bisseck   17     germany       61         82  fc köln  500.0   

      Wage Preferred Foot  International Reputation  Skill Moves Position  \
2122   1.0          right                       1.0          2.0       cb   
2123   1.0          right                       1.0          2.0       cb   

        Height  Weight  Release Clause  
2122  6.333333   183.0          1300.0  
2123  6.333333   183.0          1300.0

По столбцу ID выявлено 4 полных дубликата.

In [221]:
df["Name"].nunique()

2371

В столбце Name из 2399 значений, 28 неуникальны.

In [222]:
# находим дубликаты по столбцу Name.
# df[df.Name.duplicated()]

In [223]:
df[df['Name'] == 's. sinclair']

ID         Name  Age Nationality  Overall  Potential            Club  \
156  169214  s. sinclair   29     england       75         75          celtic   
191  172680  s. sinclair   30     england       60         60  bristol rovers   

      Value  Wage Preferred Foot  International Reputation  Skill Moves  \
156  6500.0  48.0          right                       2.0          4.0   
191   190.0   2.0          right                       1.0          2.0   

    Position    Height  Weight  Release Clause  
156       lm  5.833333   154.0         11700.0  
191       cm  5.583333   148.0           333.0

По столбцу Name дубликаты показали одинаковые имена разные люди (однофамильцы).

#### Удаляем дубликты.

In [224]:
df= df.drop('ID', axis=1).drop_duplicates()
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2395 entries, 0 to 2398
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Name                      2395 non-null   object 
 1   Age                       2395 non-null   int64  
 2   Nationality               2395 non-null   object 
 3   Overall                   2395 non-null   int64  
 4   Potential                 2395 non-null   int64  
 5   Club                      2395 non-null   object 
 6   Value                     2395 non-null   float64
 7   Wage                      2395 non-null   float64
 8   Preferred Foot            2395 non-null   object 
 9   International Reputation  2395 non-null   float64
 10  Skill Moves               2395 non-null   float64
 11  Position                  2395 non-null   object 
 12  Height                    2395 non-null   float64
 13  Weight                    2395 non-null   float64
 14  Release 

In [225]:
df.head()

Name  Age  Nationality  Overall  Potential                 Club  \
0  g. buffon   40        italy       88         88  paris saint-germain   
1   casillas   37        spain       82         82             fc porto   
2      braga   35     portugal       70         70              cd aves   
3  a. robben   34  netherlands       84         84    fc bayern münchen   
4  j. ibehre   35      england       61         61     cambridge united   

     Value   Wage Preferred Foot  International Reputation  Skill Moves  \
0   4000.0   77.0          right                       4.0          1.0   
1   1500.0   10.0           left                       4.0          1.0   
2    750.0    5.0          right                       1.0          3.0   
3  15500.0  110.0           left                       4.0          4.0   
4    120.0    2.0          right                       1.0          2.0   

  Position    Height  Weight  Release Clause  
0       gk  6.333333   203.0          7400.0  
1     none  6.083333   185.0          3000.0  
2      lcm  5.916667   165.0          1600.0  
3       rm  5.916667   176.0         25600.0  
4       st  6.333333   203.0           210.0

### Задача №3. Напишите функцию, которая добавит колонку с разбиением возраста по группам: до 20, от 20 до 30, от 30 до 36 и старше 36. Посчитайте количество футболистов в каждой категории.

По описательной статистике возраст футболистов 16-44 лет.  
Возрастные группы:  
teenage - до 20 лет  
young_adult -от 20 до 30  
adult -от 30 до 36  
older_adult -старше 36

In [227]:
df.loc[df['Age']<=20, 'Age_group'] = 'teenage'
df.loc[df['Age'].between(20,30), 'Age_group'] = 'young_adult'
df.loc[df['Age'].between(30,36), 'Age_group'] = 'adult'
df.loc[df['Age']>36, 'Age_group'] = 'older_adult'

In [228]:
df.head()

Name  Age  Nationality  Overall  Potential                 Club  \
0  g. buffon   40        italy       88         88  paris saint-germain   
1   casillas   37        spain       82         82             fc porto   
2      braga   35     portugal       70         70              cd aves   
3  a. robben   34  netherlands       84         84    fc bayern münchen   
4  j. ibehre   35      england       61         61     cambridge united   

     Value   Wage Preferred Foot  International Reputation  Skill Moves  \
0   4000.0   77.0          right                       4.0          1.0   
1   1500.0   10.0           left                       4.0          1.0   
2    750.0    5.0          right                       1.0          3.0   
3  15500.0  110.0           left                       4.0          4.0   
4    120.0    2.0          right                       1.0          2.0   

  Position    Height  Weight  Release Clause    Age_group  
0       gk  6.333333   203.0          7400.0  older_adult  
1     none  6.083333   185.0          3000.0  older_adult  
2      lcm  5.916667   165.0          1600.0        adult  
3       rm  5.916667   176.0         25600.0        adult  
4       st  6.333333   203.0           210.0        adult